framework/Samplers/Sampler.py

```diff
@@ -792,6 +792,18 @@ def localFinalizeActualSampling(self,jobObject,model,myInput):
     """
     pass
 
+  def _reassignSampledVarsPbToFullyCorrVars(self):
+    """
+      Method to reassign sampledVarsPb to the fully correlated variables
+      @ In, None
+      @ Out, None
+    """
+    fullyCorrVars = {s: self.inputInfo['SampledVarsPb'].pop(s) for s in self.inputInfo['SampledVarsPb'].keys() if "," in s}
```

Comment:
should it be:
fullyCorrVars = {s: self.inputInfo['SampledVarsPb'].pop(s) for s in self.inputInfo['SampledVarsPb'].keys() if ","  **NOT** in s}
no? you are popping all correlated vars

Reply:
It is correct.
I want to pop out the correlated bundle to assign to each correlated variable the sample SampledVarsPb.
E.g.
Instead of
```SampledVarsPb {'x1,x2,x3': 0.37061844569526153}``` we have ```SampledVarsPb {'x1': 0.37061844569526153,',x2': 0.37061844569526153,',x3': 0.37061844569526153}```

I need to do this at the end of the generateInput because otherwise we do not correctly compute the PointProbability (we do not want to consider multiple time the same pb)

Reply:
you are correct